IA-2042 Polio weekly email relax precondition for campaign

olethanh opened the pull request:
Relax the requirement for sending the reminded e-mail as some mail were not send while the users were expecting them

Remove the onset_at and virus requirement as they are often not set.


Related JIRA tickets : IA-2042

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [NA] My migrations file are included
- [No] Are there enough tests

## Changes



## How to test

Need to have the proper polio setup with mail.
I have adapted the test_send_email command to replicate the logic of the weekly_email and loop through all the campaigns

olethanh commented on hat/assets/js/apps/Iaso/domains/instances/components/InstancesFiltersComponent.js:
```diff
@@ -136,111 +77,213 @@ const InstancesFiltersComponent = ({
         [setFormState, dispatch],
     );
-
+    const startPeriodError = useMemo(() => {
+        if (formState.startPeriod?.value && formState.periodType?.value) {
+            return !isValidPeriod(
+                formState.startPeriod.value,
+                formState.periodType.value,
+            );
+        }
+        return false;
+    }, [formState.startPeriod, formState.periodType]);
+    const endPeriodError = useMemo(() => {
```
wouldn't it make more sense to use setFieldErrors and to set it in the handleFormChange?

olethanh commented on hat/assets/js/apps/Iaso/domains/instances/constants.js:
```diff
@@ -26,7 +26,8 @@ const PrettyPeriod = ({ value }) => {
 
 const LinkToForm = ({ formId, formName }) => {
-    const user = useSelector(state => state.users.current);
+    const user = 
```
seems like a merge error, this broke the feature

quang-le commented:
QA test in Jira: https://bluesquare.atlassian.net/browse/IA-936?atlOrigin=eyJpIjoiZWY4ODM2YjVlNmNjNDZkMmJmODVmYzcwYTY3ZTZlNTEiLCJwIjoiaiJ9

quang-le reviewed (changes requested):
See [Jira ticket](https://bluesquare.atlassian.net/browse/IA-936?atlOrigin=eyJpIjoiZWY4ODM2YjVlNmNjNDZkMmJmODVmYzcwYTY3ZTZlNTEiLCJwIjoiaiJ9)

Blocking and in scope (number refers to ticket description): 6
In scope but not critical: 5
Not in scope of this PR, but related to submissions: 1,2,3,4

olethanh committed:
IA-2042 Polio weekly email relax precondition for campaign

Relax the requirement for sending the reminded e-mail as some mail were not send while the users were expecting them

Remove the onset_at and virus requirement as they are often not set.

olethanh committed:
Adapt test command
